app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -203,8 +203,15 @@ case class EvaluationResults(
       case Sum(FormCtx(formComponentId))    => calculateSum(formComponentId, recData, unsupportedOperation("Number")(expr))
       case Sum(field1) =>
         loop(field1) match {
-          case lrs: ListResult => lrs.list.fold(NumberResult(0)) { case (a, b) => a + b }
-          case _               => unsupportedOperation("Number")(expr)
+          case lrs: ListResult =>
+            lrs.list.fold(NumberResult(0)) { case (a, b) =>
+              val roundedB = field1 match {
+                case Typed(_, tpe) => evalTyped(b, tpe)
+                case _             => b
+              }
+              a + roundedB
+            }
+          case _ => unsupportedOperation("Number")(expr)
```

Comment:
This is weird. Why is rounding logic living in the branch of code where we are adding numbers together? This is not systematic solution to the problem.